File: Source/AlphaTab/Rendering/Glyphs/PickStrokeGlyph.cs
using AlphaTab.Model;
using AlphaTab.Platform;

namespace AlphaTab.Rendering.Glyphs
{
    internal class PickStrokeGlyph : MusicFontGlyph
    {
        public PickStrokeGlyph(float x, float y, PickStroke pickStroke)
            : base(x, y, 0.75f, GetSymbol(pickStroke))
        {
        }

        public override void DoLayout()
        {
            Width = 9 * Scale;
            Height = 10 * Scale;
        }

        public override void Paint(float cx, float cy, ICanvas canvas)
        {
            base.Paint(cx, cy + Height, canvas);
        }

        private static MusicFontSymbol GetSymbol(PickStroke pickStroke)
        {
            switch (pickStroke)
            {
                case PickStroke.Up: return MusicFontSymbol.PickStrokeUp;
                case PickStroke.Down: return MusicFontSymbol.PickStrokeDown;
                default: return MusicFontSymbol.None;
            }
        }
    }
}

File: packages/alphatab/src/rendering/glyphs/PickStrokeGlyph.ts
import { EngravingSettings } from '@coderline/alphatab/EngravingSettings';
import { MusicFontSymbol } from '@coderline/alphatab/model/MusicFontSymbol';
import { PickStroke } from '@coderline/alphatab/model/PickStroke';
import { MusicFontGlyph } from '@coderline/alphatab/rendering/glyphs/MusicFontGlyph';

/**
 * @internal
 */
export class PickStrokeGlyph extends MusicFontGlyph {
    public constructor(x: number, y: number, pickStroke: PickStroke) {
        super(x, y, EngravingSettings.GraceScale, PickStrokeGlyph._getSymbol(pickStroke));
        this.center = true;
    }

    public override doLayout(): void {
        super.doLayout();
        this.offsetY = this.height;
    }

    private static _getSymbol(pickStroke: PickStroke): MusicFontSymbol {
        switch (pickStroke) {
            case PickStroke.Up:
                return MusicFontSymbol.StringsUpBow;
            case PickStroke.Down:
                return MusicFontSymbol.StringsDownBow;
            default:
                return MusicFontSymbol.None;
        }
    }
}
